hyperframe/frame.py

```diff
@@ -381,20 +391,21 @@ def __init__(self, stream_id=0, settings=None, **kwargs):
         self.settings = settings or {}
 
     def serialize_body(self):
-        settings = [struct.pack("!HL", setting & 0xFF, value)
-                    for setting, value in self.settings.items()]
-        return b''.join(settings)
+        return b''.join([_STRUCT_HL.pack(setting & 0xFF, value)
+                         for setting, value in self.settings.items()])
```

Comment:
Square braces here are actually faster, `.join()` with a generator is provably slower than `.join()` with a list. I don't know exactly why, but I think it has something to do with `.join()` building a list internally anyway so using a generator doesn't help.